#VRML_SIM R2023b utf8

EXTERNPROTO "../protos/Markers/WallMarker.proto"
EXTERNPROTO "../protos/Testing/TestCamera.proto"

WorldInfo {
  basicTimeStep 8
}
Viewpoint {
  orientation 0 1 0 0.5250363
  position -12 0 7.95
  ambientOcclusionRadius 0.3
  bloomThreshold 100
}
DEF SUN DirectionalLight {
  color 1 1 1
  direction 0 0 -1
  intensity 1
  castShadows TRUE
}
DEF AMBIENT Background {
  skyColor [
    0.4 0.4 0.4
  ]
  luminosity 1.6
}

Solid {
  translation 0 0 -0.002
  children [
    DEF FLOOR Shape {
      appearance PBRAppearance {
        roughness 1
        metalness 0
      }
      geometry Plane {
        size 4 20
      }
    }
  ]
  name "Floor"
  boundingObject USE FLOOR
}

Robot {
  translation 0 0 1
  children [
    # Position testing
    TestCamera {
      name "camera-marker-straight-ahead"
      translation 0 0 0
    }
    TestCamera {
      name "camera-marker-slightly-left"
      translation 0 -0.2 0
    }
    TestCamera {
      name "camera-marker-slightly-right"
      translation 0 0.2 0
    }
    TestCamera {
      name "camera-marker-slightly-closer"
      translation 0.2 0 0
    }
    TestCamera {
      name "camera-marker-slightly-further"
      translation -0.2 0 0
    }
    TestCamera {
      name "camera-marker-slightly-up"
      translation 0 0 0.2
    }
    TestCamera {
      name "camera-marker-slightly-down"
      translation 0 0 -0.2
    }
    # Orientation testing
    TestCamera {
      name "camera-marker-turned-right"
      translation 0 -2 0
    }
    TestCamera {
      name "camera-marker-turned-left"
      translation 0 -3 0
    }
    TestCamera {
      name "camera-marker-leaning-back"
      translation 0 -4 0
    }
    TestCamera {
      name "camera-marker-leaning-forwards"
      translation 0 -5 0
    }
    TestCamera {
      name "camera-marker-leaning-left"
      translation 0 -6 0
    }
    TestCamera {
      name "camera-marker-leaning-right"
      translation 0 -7 0
    }
  ]
  name "test-supervisor"
  model "TestSupervisor"
  description "A supervisor 'robot' to run integration tests."
  controller "test_supervisor"
  supervisor TRUE
}

Solid {
  children [
    # Position testing
    WallMarker {
      translation 1 0 1
      rotation 0 0 1 0
      name "A2"
      model "F2"
      texture_url [
        "../textures/arena-markers/2.png"
      ]
    }
    # Orientation testing
    WallMarker {
      translation 1 -2 1
      rotation 0 0 1 0.785398163
      name "A10"
      model "F10"
      texture_url [
        "../textures/arena-markers/10.png"
      ]
    }
    WallMarker {
      translation 1 -3 1
      rotation 0 0 1 -0.785398163
      name "A11"
      model "F11"
      texture_url [
        "../textures/arena-markers/11.png"
      ]
    }
    WallMarker {
      translation 1 -4 1
      rotation 0 1 0 0.785398163
      name "A12"
      model "F12"
      texture_url [
        "../textures/arena-markers/12.png"
      ]
    }
    WallMarker {
      translation 1 -5 1
      rotation 0 1 0 -0.785398163
      name "A13"
      model "F13"
      texture_url [
        "../textures/arena-markers/13.png"
      ]
    }
    WallMarker {
      translation 1 -6 1
      rotation 1 0 0 -0.785398163
      name "A14"
      model "F14"
      texture_url [
        "../textures/arena-markers/14.png"
      ]
    }
    WallMarker {
      translation 1 -7 1
      rotation 1 0 0 0.785398163
      name "A15"
      model "F15"
      texture_url [
        "../textures/arena-markers/15.png"
      ]
    }
  ]
  name "Markers"
}
